webview/platforms/edgechromium.py

```diff
@@ -0,0 +1,204 @@
+        self.tmpdir = None
+        self.url = None
+        self.ishtml = False
+
+        if window.html or 'localhost' in window.real_url or '127.0.0.1' in window.real_url:
+            _allow_localhost()
+
+        if window.real_url:
+            self.load_url(window.real_url)
+        elif window.html:
+            self.load_html(window.html, '')
+        else:
+            self.load_html(default_html, '')
+
+    def evaluate_js(self, script, callback=None):
+        def _callback(result):
+            if callback is None:
+                self.js_result = None if result is None or result == '' else json.loads(result)
+                self.js_result_semaphore.release()
+            else: 
+                # future js callback option to handle async js method
+                callback(result)
+                self.js_result = None
+                self.js_result_semaphore.release()
+
+        self.syncContextTaskScheduler = TaskScheduler.FromCurrentSynchronizationContext()
+        try:
+            result = self.web_view.ExecuteScriptAsync(script).ContinueWith(
+            Action[Task[String]](
+                lambda task: _callback(json.loads(task.Result))
+            ),
+            self.syncContextTaskScheduler)
+        except Exception as e:
+            logger.exception('Error occurred in script')
+            self.js_result = None
+            self.js_result_semaphore.release()
+
+    def get_current_url(self):
+        return self.url
+
+    def load_html(self, html, base_uri):
+        self.tmpdir = tempfile.mkdtemp()
+        self.temp_html = os.path.join(self.tmpdir, 'index.html')
+
+        with open(self.temp_html, 'w', encoding='utf-8') as f:
+            f.write(inject_base_uri(html, base_uri))
+
+        if self.httpd:
+            self.httpd.shutdown()
+
+        url = resolve_url('file://' + self.temp_html, True)
+        self.ishtml = True
+        self.web_view.Source = Uri(url)
+
+    def load_url(self, url):
+        self.ishtml = False
+        self.web_view.Source = Uri(url)
+
+    def on_script_notify(self, _, args):
+        try:
+            func_name, func_param, value_id = json.loads(args.get_WebMessageAsJson())
+
+            if func_name == 'alert':
+                WinForms.MessageBox.Show(func_param)
+            elif func_name == 'console':
+                print(func_param)
+            else:
+                js_bridge_call(self.pywebview_window, func_name, func_param, value_id)
+        except Exception as e:
+            logger.exception('Exception occured during on_script_notify')
+
+    def on_new_window_request(self, _, args):
+        args.set_Handled(True)
+        #webbrowser.open(str(args.get_Uri()))
+
+    def on_webview_ready(self, sender, args):
+        sender.CoreWebView2.NewWindowRequested += self.on_new_window_request
+        settings = sender.CoreWebView2.Settings
+        if not _debug: settings.AreDefaultContextMenusEnabled = False
```

Comment:
settings.AreDefaultContextMenusEnabled = _debug

Reply:
Settings tweaked for _debug.